src/apps/backend/modules/access-token/access-token-service.ts

```diff
@@ -1,10 +1,69 @@
-import AccessTokenWriter from './internal/access-token-writer';
-import { AccessToken, CreateAccessTokenParams } from './types';
+import jsonwebtoken from 'jsonwebtoken';
+
+import AccountService from '../account/account-service';
+import { ConfigService } from '../config';
+
+import {
+  AccessToken,
+  AccessTokenExpiredError,
+  AccessTokenInvalidError,
+  AccessTokenPayload,
+  CreateAccessTokenParams,
+  VerifyAccessTokenParams,
+} from './types';
 
 export default class AccessTokenService {
   public static async createAccessToken(
     params: CreateAccessTokenParams,
   ): Promise<AccessToken> {
-    return AccessTokenWriter.createAccessToken(params);
+    const account = await AccountService.getAccountByUsernamePassword({
+      password: params.password,
+      username: params.username,
+    });
+
+    const jwtSigningKey = ConfigService.getValue<string>('accounts.tokenKey');
+    const jwtExpiry = ConfigService.getValue<string>('accounts.tokenExpiry');
+
+    const jwtToken = jsonwebtoken.sign(
+      { accountId: account.id },
+      jwtSigningKey,
+      { expiresIn: jwtExpiry },
+    );
+
+    const accessToken = new AccessToken();
+    accessToken.accountId = account.id;
+    accessToken.token = jwtToken;
+
+    const jwtTokenDecoded = jsonwebtoken.decode(jwtToken) as jsonwebtoken.JwtPayload;
+    accessToken.expiresAt = new Date(jwtTokenDecoded.exp * 1000);
+
+    return accessToken;
+  }
+
+  public static verifyAccessToken(
+    params: VerifyAccessTokenParams,
+  ): AccessTokenPayload {
+    const { token } = params;
+    let verifiedToken: jsonwebtoken.JwtPayload;
+
+    try {
+      const jwtSigningKey = ConfigService.getValue<string>('accounts.tokenKey');
+
+      verifiedToken = jsonwebtoken.verify(
+        token,
+        jwtSigningKey,
+        { ignoreExpiration: true },
+      ) as jsonwebtoken.JwtPayload;
+    } catch (e) {
+      throw new AccessTokenInvalidError();
+    }
+
+    if (verifiedToken.exp * 1000 < Date.now()) {
+      throw new AccessTokenExpiredError();
+    }
```

Comment:
The .verify also throws an error if the token has expired

Reply:
Yes it does, but we still need to massage to one of our application errors in order to propagate it further up.

Reply:
ok. verify throws error with a specific code. the if block can be moved there too. But yeah, `if` block is needed there too.